ui/src/shared/apis/query.ts

```diff
@@ -1,113 +1,73 @@
-import _ from 'lodash'
 import {getDeep} from 'src/utils/wrappers'
-import {
-  handleSuccess,
-  handleError,
-  handleLoading,
-} from 'src/shared/actions/timeSeries'
 import {analyzeQueries} from 'src/shared/apis'
-import {DEFAULT_DURATION_MS} from 'src/shared/constants'
+import {TEMP_VAR_INTERVAL, DEFAULT_DURATION_MS} from 'src/shared/constants'
 import replaceTemplates, {replaceInterval} from 'src/tempVars/utils/replace'
 import {proxy} from 'src/utils/queryUrlGenerator'
 
-import {Source} from 'src/types'
-
-import {Template} from 'src/types'
-
-const noop = () => ({
-  type: 'NOOP',
-  payload: {},
-})
+import {Source, Template} from 'src/types'
+import {TimeSeriesResponse} from 'src/types/series'
 
+// REVIEW: why is this different than the `Query` in src/types?
```

Comment:
My guess is that this concept of a "query" relates to the UI. It is checking what the user has selected to decide whether or not it can display a complete query string from the builder

Not sure though!